catalog/task/update-ocp-tag/0.1/update-ocp-tag.yaml

```diff
@@ -0,0 +1,57 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: update-ocp-tag
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task to update pull-spec tag with
+    valid OCP version from get-ocp-version task.
+  params:
+    - name: fromIndex
+      description: The source Index image (catalog of catalogs) FBC fragment
+      type: string
+    - name: targetIndex
+      description: Index image (catalog of catalogs) the FBC fragment will be added to
+      type: string
+    - name: binaryImage
+      description: OCP binary image to be baked into the index image
+      type: string
+    - name: ocpVersion
+      description: Stored OCP version from get-ocp-version task result
+      type: string
+  results:
+    - name: updated-fromIndex
+      description: source Index image (catalog of catalogs) FBC fragment with updated tag
+    - name: updated-targetIndex
+      description: Index image (catalog of catalogs) the FBC fragment will be added to with updated tag
+    - name: updated-binaryImage
+      description: OCP binary image to be baked into the index image with updated tag
+  steps:
+    - name: update-ocp-tag
+      image: quay.io/hacbs-release/release-utils:4d8649dbb2b626f5fe9f4ff83c1bc3be268fad31
+      script: |
+       #!/usr/bin/env sh
+
+        # Function to replace tag in an image
+        replace_tag() {
+            local image="${1}"
+            local oldTag=$(echo "$image" | sed -E 's/.*([vV][0-9]+\.[0-9]{2})$/\1/')
+            local updatedImage="${1%:*}:$(params.ocpVersion)"
+            echo "$updatedImage"
+        }
+
+        # Access the updated image
+        updatedFromIndex=$(replace_tag "$(params.fromIndex)")
+        updatedTargetIndex=$(replace_tag "$(params.targetIndex)")
+        updatedBinaryImage=$(replace_tag "$(params.binaryImage)")
+
+        echo "Updated values"
+        echo "$updatedFromIndex" | tee $(results.updated-fromIndex.path)
+        echo "$updatedTargetIndex" | tee $(results.updated-targetIndex.path)
+        echo "$updatedBinaryImage" | tee $(results.updated-binaryImage.path)
```

Comment:
use `echo -n` instead as it is adding "\n"  to the result and breaking the pipelinerun.
```json
- name: params
      value: >-
        [{"name":"binaryImage","value":"registry.redhat.io/openshift4/ose-operator-registry:v4.12

        "},{"name":"fromIndex","value":"quay.io/scoheb/fbc-index-testing:v4.12

        "},
```

Reply:
Done, thanks for pointing out @theflockers

Reply:
So, it works fine now but the output is all in the same line (as `tee` only prints out what it receives).
In the sense of functionality it is all fine, but I guess that we want a readable output. 

Maybe change it to a regular `echo -n XXX > $(path.to.result)`  and go back adding the `set -x`? @mmalina @johnbieren

Reply:
Yeah, that, or you could also just add a plain `echo` after each line.

Reply:
Thanks for the input @theflockers and @mmalina 
updated.